crates/biome_yaml_parser/src/token_source.rs

```diff
@@ -24,15 +24,15 @@ impl<'source> YamlTokenSource<'source> {
         let lexer = BufferedLexer::new(lexer);
 
         let mut source = YamlTokenSource::new(lexer);
-        source.next_non_trivia_token(true);
+        source.next_non_trivia_token(true, YamlLexContext::Regular);
         source
     }
 
-    fn next_non_trivia_token(&mut self, first_token: bool) {
+    fn next_non_trivia_token(&mut self, first_token: bool, context: YamlLexContext) {
```

Comment:
nit: the equivalent for this in the html parser has these arguments reversed: `context: HtmlLexContext, first_token: bool` and it would be better if we stay consistent. (so if the html parser is actually the one at fault we would change it to match the other languages, of course)